prometheus/vec.go

```diff
@@ -80,6 +80,16 @@ func (m *MetricVec) DeleteLabelValues(lvs ...string) bool {
 	return m.metricMap.deleteByHashWithLabelValues(h, lvs, m.curry)
 }
 
+// DeletePartialMatchLabelValues deletes all metrics where the variable labels
+// contain all of the values passed. The order of the passed values does not matter.
+// It returns the number of metrics deleted.
+
+// This method deletes a metric if the given value is present for any label.
+// To delete a metric based on a partial match for a particular label, use DeletePartialMatch().
+func (m *MetricVec) DeletePartialMatchLabelValues(lvs ...string) int {
```

Comment:
shouldn't we have label name passed for it too?

Reply:
Do we have use case when we don't know label name? This seems tedious for implementation and prone to errors 🤔

Reply:
You're right, I had in my head that deleting with unknown label names would also be needed but I don't have a use case for it personally. I'm fine to drop it and support only `DeletePartialMatch`. Alternatively, `DeletePartialMatchLabelValues` could be made to delete based on the presence of any of the given values for a particular label as you described, if that's desired behavior.

Reply:
I dropped the `DeletePartialMatchLabelValues` implementation -- if someone finds it to be valuable then we can re-introduce it